tests/pocket_test.cpp

```diff
@@ -0,0 +1,832 @@
+    REQUIRE( sonic.length() < sword.length() );
+
+    // Sheath just long enough for sonic screwdriver
+    pocket_data data_sheath( item_pocket::pocket_type::CONTAINER );
+
+    // Has plenty of weight/ volume, since we're only testing length
+    data_sheath.volume_capacity = 10_liter;
+    data_sheath.max_contains_weight = 10_kilogram;
+
+    GIVEN( "pocket has max_item_length enough for some items, but not others" ) {
+        data_sheath.max_item_length = sonic.length();
+        item_pocket pocket_sheath( &data_sheath );
+
+        THEN( "it can contain an item with longest_side shorter than max_item_length" ) {
+            expect_can_contain( pocket_sheath, screwdriver );
+        }
+
+        THEN( "it can contain an item with longest_side equal to max_item_length" ) {
+            expect_can_contain( pocket_sheath, sonic );
+        }
+
+        THEN( "it cannot contain an item with longest_side longer than max_item_length" ) {
+            expect_cannot_contain( pocket_sheath, sword, "item is too long",
+                                   item_pocket::contain_code::ERR_TOO_BIG );
+        }
+    }
+}
+
+// Max item volume
+// ---------------
+// "max_item_volume" is the size of the mouth or opening into the pocket. Liquids and gases, as well
+// as "soft" items, can always fit through. All other items must fit within "max_item_volume" to be
+// put in the pocket.
+//
+// Related JSON fields:
+// "max_item_volume"
+//
+// Functions:
+// item_pocket::can_contain
+//
+TEST_CASE( "max item volume", "[pocket][max_item_volume]" )
+{
+    // Test items
+    item screwdriver( "test_screwdriver" );
+    item water( "water" );
+    item rag( "test_rag" );
+    item rock( "test_rock" );
+
+    // Water-tight jug-style container
+    pocket_data data_jug( item_pocket::pocket_type::CONTAINER );
+    data_jug.watertight = true;
+
+    // No significant limits on length or weight
+    data_jug.max_item_length = 1_meter;
+    data_jug.max_contains_weight = 10_kilogram;
+
+    // Jug capacity is more than enough for several rocks, rags, or screwdrivers
+    data_jug.volume_capacity = 2_liter;
+    REQUIRE( data_jug.volume_capacity > 5 * rock.volume() );
+    REQUIRE( data_jug.volume_capacity > 5 * rag.volume() );
+    REQUIRE( data_jug.volume_capacity > 10 * screwdriver.volume() );
+
+    // But it has a narrow mouth, just barely enough to fit a screwdriver
+    data_jug.max_item_volume = screwdriver.volume();
+    // And the rock is definitely too big to fit
+    REQUIRE( rock.volume() > 2 * screwdriver.volume() );
+
+    GIVEN( "large watertight pocket with a narrow opening" ) {
+        item_pocket pocket_jug( &data_jug );
+
+        THEN( "it can contain liquids" ) {
+            expect_can_contain( pocket_jug, water );
+        }
+        THEN( "it can contain solid items that fit through the opening" ) {
+            expect_can_contain( pocket_jug, screwdriver );
+        }
+        THEN( "it can contain soft items larger than the opening" ) {
```

Comment:
following your other conventions here i expected an assertion that a rag is larger in volume than the mouth opening

Reply:
Good call - an `is_soft` verification would help too. Will add.